numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -701,6 +720,10 @@ typedef struct tagPyArrayObject_fields {
     int flags;
     /* For weak references */
     PyObject *weakreflist;
+    /*
+     * For alloc/malloc/realloc/free/memcpy per object
+     */
+    PyDataMem_Handler *mem_handler;
```

Comment:
Do you have any thoughts regarding your comment here? https://github.com/numpy/numpy/pull/16938#discussion_r459873101

I honestly do not know if there is much we can do, but I guess there may be some small things.